processing/src/main/java/io/druid/segment/filter/OrFilter.java

```diff
@@ -103,23 +104,14 @@ public ValueMatcher makeMatcher(
       matchers.add(0, offsetMatcher);
     }
 
-    return new ValueMatcher()
-    {
-      @Override
-      public boolean matches()
-      {
-        for (ValueMatcher valueMatcher : matchers) {
-          if (valueMatcher.matches()) {
-            return true;
-          }
-        }
-        return false;
-      }
-    };
+    return makeMatcher(matchers.toArray(new ValueMatcher[0]));
   }
 
 
   private ValueMatcher makeMatcher(final ValueMatcher[] baseMatchers){
+    if (baseMatchers.length == 0) {
+      return BooleanValueMatcher.of(false);
```

Comment:
It's not intuitive. ```baseMatchers``` will have 0 length only when ```filters``` is an empty list. OrFilter should not filter anything in this case.
In addition, when does ```baseMatchers``` become a zero-length array? The empty filter is not allowed in the constructor.